src/compiler/types.ts

```diff
@@ -6185,6 +6185,8 @@ export const enum TypeFlags {
     StringMapping   = 1 << 28,  // Uppercase/Lowercase type
     /** @internal */
     Reserved1       = 1 << 29,  // Used by union/intersection type construction
+    /** @internal */
+    Reserved2       = 1 << 30,  // Used by union/intersection type construction

```

Comment:
remind me, do we have 31 bits available?

Reply:
In modern Node, yep, we do.